README_WAKE.adoc

```diff
@@ -0,0 +1,154 @@
+= Rocket Chip Generator
+
+This repository contains the Rocket chip generator necessary to instantiate
+the RISC-V Rocket Core. For more information on Rocket Chip, please consult our
+http://www.eecs.berkeley.edu/Pubs/TechRpts/2016/EECS-2016-17.html[technical report].
+
+== Dependencies
+
+This repository uses version 0.10.1 of https://github.com/sifive/wit[Wit] and version 0.15.1 of https://github.com/sifive/wake[Wake].
+Please see the respective repositories and their releases to install these dependencies.
+
+== Checkout The Code
+
+This repository uses https://github.com/sifive/wit[Wit] for dependency fetching and resolution.
+Please see the https://github.com/sifive/wit/blob/v0.10.1/share/doc/wit/tutorial.md[Wit tutorial] for more information.
+
+----
+$ wit init workspace -a git@github.com:chipsalliance/rocket-chip.git
+----
+
+This will clone rocket-chip as well as resolve and fetch all dependencies.
+
+== Compile Rocket Chip
+
+It is not necessary to compile separately from invocation, but it might be useful to understand how the different pieces work.
+
+This repository uses https://github.com/sifive/wake[Wake] for orchestrating the build.
+We strongly recommend checking out the https://github.com/sifive/wake/blob/v0.15.1/share/doc/wake/tutorial.md[Wake tutorial].
+You can find the Rocket Chip Wake rules in `build.wake` within this repository.
+
+The Rocket Chip Scala code is compiled using https://github.com/sifive/api-scala-sifive[API Scala SiFive],
+and can thus be compiled with:
+
+----
+$ wake compileScalaModule rocketchipScalaModule
+----
+
+The return value of that function is the `Path` to the compile jar.
+
+== Run Rocket Chip Generator
+
+=== API
+
+You can run the generator with the Wake rule `runRocketChipGenerator`.
+This function accepts a single `RocketChipGeneratorOptions` as an argument.
+
+----
```

Comment:
Although GitHub doesn't have a built-in Wake syntax highlighter, if we ever produce static HTML versions of these READMEs it'll be nicer to explicitly annotate the code block with the language so that we can apply the appropriate syntax highlighting.

See how I did this in the Wake documentation that I wrote recently: https://github.com/sifive/wake/blame/2dba999205b58ace3dc7d92a1010335f201dff58/share/doc/wake/from-scala.adoc#L40

Reply:
Oh good call, I'll add that

Reply:
I do think we should look into generating websites sooner rather than later, easier than trying to get Github to accept Wake as a language and then we can use the more advanced features (like pulling in code rather than copy pasting it which I am only currently doing out of necessity.